### ISSUE TITLE
Reasoning for -stdlib=libc++ flag?

### ISSUE THREAD

**Issue by qqii** [tizian/tonemapper#6]
I'm building your project on Linux and noticed you've got [`-stdlib=libc++` in CMakeLists.txt#L14](https://github.com/tizian/tonemapper/blob/master/CMakeLists.txt#L14). 

Reading your README.md, I'm assuming you're developing on MacOS, which seems to be backed up by this [stackoverflow reply](https://stackoverflow.com/a/19774902). 

I'm not that familiar with CMAKE or clang, so I was wondering if there a reason that you're explicitly specifying it [against the advice of this SO answer](https://stackoverflow.com/a/50407611)?

**Comment by tizian** [comment]
This is just the way I setup up CMake 5 years ago, and probably didn't think much of it (and I certainly didn't try to "go against" some online discussion...).
Looking at this now I guess the flag is pretty clang specific, but I just never tried to compile this code with gcc. It looks like even in that Stackoverflow thread recommends `libc++` together with clang, right?

**Comment by qqii** [comment]
Sorry if I came off as aggressive to you. I'm re-reading the SO post I linked and as a non-mac user I'm more confused than when I originally opened this issue. 

I originally curious if you were maybe dependant on something `libc++` specific or other standard implementations versions didn't follow some spec. Since then I've managed to compile it fine with `g++` on Linux. Reading both the top and second answer the issue seems to be:

On OS X before Mavericks `g++` is an alias for `clang++` and by default uses `-stdlib=libstdc++`, the version of `libstdc++` that apple shipped is old and does not support C++11.

Therefore the solution for using C++11 on OS X before Mavericks was `-std=c++11 -stdlib=libc++`, and after OS X Mavericks there's no need to specify `-stdlib` at all.
